packages/genesys-spark-components/src/components/stable/gux-table/example.html

```diff
@@ -970,7 +985,21 @@ <h2>Data table with fixed columns</h2>
       }
 
 
-    })
+    });
+
+  const selectAll = document.getElementById('selectAll');
+  const deselectAll = document.getElementById('deselecAll');
+  const allRowSelect = document.getElementById('allRowSelect');
+
+  selectAll.addEventListener('click', () => {
+    allRowSelect.setIndeterminate(false);
```

Comment:
Would it be better to house this within the `setChecked` method in the shadowDOM like,


```
async setChecked(checked: boolean): Promise<void> {
    this.selected = checked;
   this.setIndeterminate(false);
    this.internalallrowselectchange.emit(this.selected);
  }
```
or
```
async setChecked(checked: boolean): Promise<void> {
    this.selected = checked;
   this.inputElement.indeterminate = false
    this.internalallrowselectchange.emit(this.selected);
  }
```
as I am not sure is it valid to call a method with a Watch decorator internally.

Reply:
Yeah I did initially look at that but decided to keep the setting of checked & indeterminate states separate to keep it inline with the native checkbox as there were a few issues.

Apparently 'indeterminate' is a presentational state, just visual,  while the value of a checkbox is either checked or unchecked. I thought it best to stick with this to avoid any unforeseen issues down the road by diverging from the native checkbox.